{
  delayTestFinish(DELAY_TEST_FINISH);
  final RuntimeException exception1=new RuntimeException("first exception");
  final RuntimeException exception2=new RuntimeException("second exception");
  final CountingReceiver count=new CountingReceiver();
  SimpleFooRequest context=req.simpleFooRequest();
  context.pleaseCrash(42).to(count);
  context.returnNullString().to(new ThrowingReceiver<String>(exception1));
  context.returnNullString().to(count);
  fireContextAndCatch(context,new ThrowingReceiver<Void>(exception2),new ExceptionVerifier(){
    @Override public void verify(    Throwable e){
      assertUmbrellaException(e,exception1,exception2);
      count.assertCounts(1,1,0);
      finishTestAndReset();
    }
  }
);
}
